client/src/components/Header/Search/Search.js

```diff
@@ -0,0 +1,188 @@
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "0px",
+    transition: "width 2s",
+    "&:focus": {
+      width: "100%",
+    },
+    // [theme.breakpoints.up("md")]: {
+    //     width: "20ch",
+    // },
+  },
+}));
+
+const Search = ({ darkMode, setDarkMode }) => {
+  const classes = useStyles();
+  const [results, setResults] = useState([]);
+  const allChallenges = useContext(AllChallenges).challenges;
+  const [searchValue, setSearchValue] = useState("");
+
+  const letterColor = {
+    color: darkMode ? "white" : "black",
+  };
+
+  //search function to the the challenges
+  const search = (e) => {
+    setSearchValue(e.target.value);
+    let { value } = e.target;
+    if (value.length === 0) {
+      setSearchValue("");
+      return setResults([]);
+    }
+    try {
+      const filteredChallenges = [];
+      for (let i = 0; i < allChallenges.length; i++) {
+        if (allChallenges[i].name.toLowerCase().includes(value.toLowerCase())) {
+          filteredChallenges.push(allChallenges[i]);
+        }
+      }
+      setResults(filteredChallenges);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const closeSearch = useCallback(() => {
+    setSearchValue("");
+    setResults([]);
+  }, []);
+
+//TODO find good practice way
+  const onSearchLooseFocus = () => {
+    setTimeout(() => {
+        setResults([]);
+        setSearchValue("");
+    }, 200)
+  };
+
+  const resultsList =
+    results &&
+    results.length > 0 &&
+    results.map((result) => {
+      return (
+        <SearchTicket
+          darkMode={darkMode}
+          ticket={result}
+          key={result.id}
+          closeSearch={closeSearch}
+        />
+      );
+    });
+  //search bar item
+  const searchInput = (
+    <div className={darkMode ? classes.search : classes.searchLight}>
+      <div className={classes.searchIcon}>
+        <SearchIcon style={letterColor} />
+      </div>
+      <InputBase
+        style={letterColor}
+        id={"searchBar"}
+        placeholder={"Search..."}
+        onChange={search}
+        autoComplete="off"
+        classes={{
+          root: classes.inputRoot,
+          input: classes.inputInput,
+        }}
+        onBlur={onSearchLooseFocus}
+        value={searchValue}
+        inputProps={{ "aria-label": "search" }}
+      />
+    </div>
+  );
+  const SearchContainerStyle = {
+    backgroundColor: darkMode ? "rgb(51,51,51)" : "white",
+    color: darkMode ? "white" : "black",
+  };
+
+  return (
+    <>
+      <div id="search">{searchInput}</div>
+      <Divider />
+      <div
+        id="searchResults"
+        className={results.length !== 0 ? "open" : "closed"}
+      >
+        <div className="display" style={SearchContainerStyle}>
+          {resultsList}
+        </div>
+      </div>
+    </>
+  );
+};
+export default Search;
```

Comment:
Break to small component + style file. "searchInput" - can be a clasic sperete component